gosdk/rpc/rpc.go

```diff
@@ -66,47 +68,157 @@ func (s *StandardRPCServer) handleRPC(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	w.Header().Set("Content-Type", "application/json")
-	w.Header().Set("Access-Control-Allow-Origin", "*")
-	w.Header().Set("Access-Control-Allow-Methods", "POST, OPTIONS")
-	w.Header().Set("Access-Control-Allow-Headers", "Content-Type")
-
-	var req JSONRPCRequest
-	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
-		s.writeError(w, -32700, "Parse error", req.ID)
+	s.setCORSHeaders(w, "POST, OPTIONS")
 
+	// Process request middlewares as early as possible, before reading body
+	if s.processRequestMiddlewares(w, r) {
 		return
 	}
 
-	if req.JSONRPC != "2.0" {
-		s.writeError(w, -32600, "Invalid Request", req.ID)
+	// Read the entire request body
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		s.writeError(w, -32700, "Parse error")
 
 		return
 	}
+	defer r.Body.Close()
 
-	handler, exists := s.customMethods[req.Method]
-	if !exists {
-		s.writeError(w, -32601, fmt.Errorf("%w: %s", ErrMethodNotFound, req.Method).Error(), req.ID)
+	// Try to decode as batch request (array) first
+	var batchReq []JSONRPCRequest
```

Comment:
What is '' was given?

Reply:
I checked, There is a test case for this scenario as well and it working